### PR TITLE
ENH : better anonymous session handling

### PR DESCRIPTION
**Pull request by Poltuu**
en l'absence de curprincipal certaines choses devraient planter plus élégamment, d'autres ne devraient pas du tout planter

### CODE REVIEW

**Comment by LuccaIntegration** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ReadOnlyRestCollection.cs#L126](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L126): Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
1. ![MAJOR][MAJOR] [ReadOnlyRestCollection.cs#L163](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L163): Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
1. ![MAJOR][MAJOR] [ReadOnlyRestCollection.cs#L197](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L197): Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
1. ![INFO][INFO] [ReadOnlyRestCollection.cs#L126](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L126): Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)
1. ![INFO][INFO] [ReadOnlyRestCollection.cs#L161](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L161): Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)
1. ![INFO][INFO] [ReadOnlyRestCollection.cs#L196](https://github.com/LuccaSA/RestDrivenDomain/blob/37d32fca0208b272c9733a306d00aa69f93d7a43/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L196): Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs [hunk 1]
```diff
@@ -109,22 +109,17 @@ public ReadOnlyRestCollection(IRepository<TEntity> repository, IExecutionContext
...
-            AttachOperationsToEntities(new List<TEntity>
-            {
-                entity
-            });
+            AttachOperationsToEntities(new List<TEntity> { entity });
         }
 
         private void AttachOperationsToEntities(IEnumerable<TEntity> entities)
         {
             var operationsForAttach = new List<Operation>(); //TODO  _appInstance.GetAllOperations<TEntity>();
```
Comment (LuccaIntegration) [inline Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs]: MAJOR Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
INFO Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)

**Comment by LuccaIntegration** [comment]
SonarQube analysis reported 6 issues
* ![MAJOR][MAJOR] 3 major
* ![INFO][INFO] 3 info

Watch the comments in this conversation to review them.

#### 4 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ReadOnlyRestCollection.cs#L158](https://github.com/LuccaSA/RestDrivenDomain/blob/c3a63da11ac04fcd7bd66c5a3a24ef6013a86585/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L158): Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
1. ![MAJOR][MAJOR] [ReadOnlyRestCollection.cs#L187](https://github.com/LuccaSA/RestDrivenDomain/blob/c3a63da11ac04fcd7bd66c5a3a24ef6013a86585/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L187): Remove this commented out code. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS125)
1. ![INFO][INFO] [ReadOnlyRestCollection.cs#L156](https://github.com/LuccaSA/RestDrivenDomain/blob/c3a63da11ac04fcd7bd66c5a3a24ef6013a86585/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L156): Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)
1. ![INFO][INFO] [ReadOnlyRestCollection.cs#L186](https://github.com/LuccaSA/RestDrivenDomain/blob/c3a63da11ac04fcd7bd66c5a3a24ef6013a86585/Domain/RDD.Domain/Models/ReadOnlyRestCollection.cs#L186): Complete the task associated to this 'TODO' comment. [rule](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by rducom** [comment]
faut rajouter du coverage, y'a pas le choix 😸

**Comment by nfaugout** [comment]
@Poltuu tu devrais faire qq T.U. pour t'assurer que ces nouvelles exceptions sont bien jetées dans les cas que tu as identifiés, ça fera plaisir à Codecov et on pourra merger cette PR.

**Comment by nfaugout** [comment]
@Poltuu elle est encore d'actualité cette PR ou elle est obsolète ?

**Comment by Poltuu** [comment]
fixé via meilleure gestion des droits

**Pull request closed without merging**